### ISSUE TITLE
Emulate quantified comparison predicates for row value expressions, if not natively supported

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#3505]
With #2054, support for quantified comparison predicates for row value expressions was added:

``` sql
(A, B) >= ANY(SELECT x, y FROM ...)
(A, B) >= ALL(SELECT x, y FROM ...)
```

This is not supported in many databases, but it can be emulated as such:

``` sql
(A, B) >= (SELECT x, y FROM ... ORDER BY 1 DESC, 2 DESC LIMIT 1)
(A, B) >= (SELECT x, y FROM ... ORDER BY 1 ASC , 2 ASC  LIMIT 1)
```

If `(A, B) >= (SELECT ..)` is already emulated using `EXISTS` or `NOT EXISTS`, then we don't need `ORDER BY .. LIMIT`. Perhaps, it would be better to always emulate the predicate with `EXISTS` or `NOT EXISTS` to prevent a potentially `O(N log N)` `ORDER BY` operation, in case the RDBMS isn't capable to turn it into `O(N)` thanks to `LIMIT 1`

... in fact, this emulation will also extend to single-column quantified comparison predicates, which can thus be emulated.

**Comment by lukaseder** [comment]
If `(A, B) >= (SELECT ..)` is already emulated using `EXISTS` or `NOT EXISTS`, then we don't need `ORDER BY .. LIMIT`. Perhaps, it would be better to always emulate the predicate with `EXISTS` or `NOT EXISTS` to prevent a potentially `O(N log N)` `ORDER BY` operation, in case the RDBMS isn't capable to turn it into `O(N)` thanks to `LIMIT 1`

**Comment by lukaseder** [comment]
We're no longer using the `ORDER BY .. LIMIT 1` emulation and instead resort to using `[ NOT ] EXISTS` in every case.

Now, about correct `NULL` handling and 3VL

**Comment by lukaseder** [comment]
While the emulations for `>`, `>=`, `<`, `<=` seem to be `NULL` safe (unless integration tests overlook something), the `=` and `!=` emulations are not yet.

**Comment by lukaseder** [comment]
Also, the current implementation using `[ NOT ] EXISTS` works only in the `WHERE` clause, where `FALSE` and `NULL` has the same effect. It does not work correctly when projecting `BOOLEAN` columns, i.e. when using `DSL.field(Condition)`

**Comment by lukaseder** [comment]
`NULL` handling will be fixed in a separate issue: #10545
